sunpy/timeseries/sources/goes.py

```diff
@@ -173,8 +174,8 @@ def _parse_hdus(cls, hdulist):
         else:
             raise ValueError("Don't know how to parse this file")
 
-        times = [start_time + datetime.timedelta(seconds=int(np.floor(s)),
-                                                    microseconds=int((s - np.floor(s)) * 1e6)) for s in seconds_from_start]
+        times = [start_time + TimeDelta(s*u.second) for s in seconds_from_start]
```

Comment:
This needs to be converted to `datetime` before it is passed to `DataFrame` on line 188 below.